Sources/Calendar.php

```diff
@@ -365,6 +423,55 @@ function CalendarPost()
 	$context['linktree'][] = array(
 		'name' => $context['page_title'],
 	);
+
+	loadCSSFile('jquery-ui.datepicker.css', array('defer' => false), 'smf_datepicker');
+	loadCSSFile('jquery.timepicker.css', array('defer' => false), 'smf_timepicker');
+	loadJavascriptFile('jquery-ui.datepicker.min.js', array('defer' => true), 'smf_datepicker');
+	loadJavascriptFile('jquery.timepicker.min.js', array('defer' => true), 'smf_timepicker');
+	loadJavascriptFile('datepair.min.js', array('defer' => true), 'smf_datepair');
+	addInlineJavascript('
+	$("#allday").click(function(){   
+		$("#start_time").attr("disabled", this.checked);
+		$("#end_time").attr("disabled", this.checked);
+		$("#tz").attr("disabled", this.checked);
+	});
+	$( "#event_time_input .date_input" ).datepicker({
+		dateFormat: "yy-mm-dd",
```

Comment:
As I noted in the PR's opening comments:

> Apart from keeping the JS happy, there's no reason the date fields couldn't use any format that PHP understands.

So, if you or someone else were to create a subsequent PR to make the JavaScript date parsing and formatting more robust, that would be lovely. So long the JS outputs a string in the input field in a format that PHP can parse, it can be whatever you like.